geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/CacheClientProxy.java

```diff
@@ -345,19 +348,15 @@ protected CacheClientProxy(InternalCache cache, CacheClientNotifier ccn, Socket
     initializeClientAuths();
   }
 
-  private void initializeClientAuths() {
-    if (AcceptorImpl.isPostAuthzCallbackPresent()) {
-      clientUserAuths = ServerConnection.getClientUserAuths(proxyID);
-    }
+  void initializeClientAuths() {
+    clientUserAuths = ServerConnection.getClientUserAuths(proxyID);
```

Comment:
One way to make the use of `CacheClientProxy` more testable is to wrap the`ServerConnection.getClientUserAuths` calls in a ` java.util.function.Function` or custom inner-interface.

You basically add a new field:
```
private final Function<ClientProxyMembershipID, ClientUserAuths> getClientUserAuthsFunction;
```
Then update the 2nd `CacheClientProxy` constructor marked with `@VisibleForTesting` to include one more parameter:
```
MessageDispatcherFactory messageDispatcherFactory,
Function<ClientProxyMembershipID, ClientUserAuths> getClientUserAuthsFunction) {
```
Change the 1st constructor to pass in a method reference:
```
DEFAULT_MESSAGEDISPATCHERFACTORY,
ServerConnection::getClientUserAuths);
```
And save the the parameter in the 2nd constructor to the field:
```
this.getClientUserAuthsFunction = getClientUserAuthsFunction;
```
And finally use it anywhere the class calls the static `ServerConnection.getClientUserAuths`:
```
clientUserAuths = getClientUserAuths.apply(proxyID);
```

Reply:
That's a useful trick, I will see if I need more tests that would utilize this.

Reply:
If I just add a constructor that takes a ClientUserAuth object as the extra parameter instead of the supplier function, would there be any difference as far as testability goes?

Reply:
If `getClientUserAuths` was non-static then you could pass an instance of it into the constructor and that would be even better. Since it's static you need to wrap it within a `Function` interface.

Reply:
PS: this was just sharing a pattern that might help with some of the testing... use it in later PRs instead of this one if you don't need it here

Reply:
Great! Thanks!